src/NUnitFramework/framework/Interfaces/ICommonTestData.cs

```diff
@@ -0,0 +1,49 @@
+// ***********************************************************************
+// Copyright (c) 2020 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+#nullable enable
+
+namespace NUnit.Framework.Interfaces
+{
+    /// <summary>
+    /// Represents data that is common between an <see cref="ITest"/>
+    /// and <see cref="ITestData"/> to allow for de-duplication of code.
+    /// </summary>
+    public interface ICommonTestData
```

Comment:
The places we wanted to de-duplicate are either an `ITest` or an `ITestData`, so I couldn't add an extension method for both of those. I pulled the common properties into this common base inteface so that we could start de-duplicating more of the messages with extension methods. I thought extension methods were a good option for common code now that we no longer have .NET 2.0 targets.

Reply:
If I remember right, moving interface methods down to a base interface causes MissingMethodException for already-compiled libraries that referenced the previous version of NUnit when they run against the new version of NUnit. E.g. https://stackoverflow.com/questions/55061714/missingmethodexception-after-extracting-base-interface.

The primary example is a NuGet test helper library that itself references on NUnit. The user installs the newest version of the helper library and the newest version of NUnit into their project, and then the helper library calls into `ITest.RunState` which existed at the point in time when the helper library was compiled. The newest version doesn't have that method though because it moved to `ICommonTestData.RunState`. So the CLR says "I can't find that." (The CLR doesn't search base interfaces because of the diamond dependency problem. At least, this was the state of things before .NET Core added Default Interface Methods.)

Less of a hassle (because it's actionable) is that this change will also cause the following code to stop compiling:

```cs
class MyTest : ITest
{
    // ...

    // The compiler will error until ITest is changed to ICommonTestData:
    RunState ITest.RunState => _runState;
}
```

What do you think about making `AddIgnoreUntilReason` an extension method on `IPropertyBag` so it's `test.Properties.AddIgnoreUntilReason`?

Reply:
That sounds like a good alternative. I'll do it tomorrow. Thanks.

Reply:
@jnm2 I've reverted the interface change and switched the extension to iPropertyBag. This is ready for review again.